test/helpers/utils.go

```diff
@@ -135,3 +136,33 @@ func Fail(description string, callerSkip ...int) {
 	}
 	ginkgo.Fail(description, callerSkip...)
 }
+
+// ReportDirectory creates and returns the directory path to export all report
+// commands that need to be run in case of fail.
+// If dir can not be created it'll return an error
+func ReportDirectory() (string, error) {
+	testDesc := ginkgo.CurrentGinkgoTestDescription()
+	testPath := fmt.Sprintf("%s%s/",
+		testResultsPath,
+		strings.Replace(testDesc.FullTestText, " ", "", -1))
+	if _, err := os.Stat(testPath); err == nil {
+		return testPath, nil
+	}
+	err := os.MkdirAll(testPath, 0777)
+	return testPath, err
+}
+
+//reportMap get a path, a map with command and output file and a node instance,
```

Comment:
Make the mapping relationship more clear; "reportMap saves the output of the commands in the filename to which they map to in reportCmds"